pilot/pkg/model/context.go

```diff
@@ -158,6 +166,49 @@ type Proxy struct {
 
 	// GlobalUnicastIP stores the globacl unicast IP if available, otherwise nil
 	GlobalUnicastIP string
+
+	// Generator is used to generate resources for the node, based on the PushContext.
+	Generator Generator
+
+	// Active contains the list of watched resources for the proxy, keyed by the DiscoveryRequest type.
+	// It is nil if the Proxy uses the default generator
+	Active map[string]*WatchedResource
+}
+
+// WatchedResource tracks an active DiscoveryRequest type.
+type WatchedResource struct {
+	// TypeUrl is copied from the DiscoveryRequest that initiated watching this resource.
+	TypeUrl string
+
+	// Generator is the generator that will generate this resource.
+	Generator Generator
```

Comment:
Why do we need this here? We have a super confusing cross referencing going on. Proxy has a generated and a watched resource which also has a generator. Generator itself has a function that takes in a watched resource which also has a generator? I am a bit lost

Reply:
Actually seems almost all of these fields are not used?

Reply:
They are used - I fixed the initial implementation that was following the old model of 'flat' fields in Xds connection struct. 

A Generator generates config for a watched resource - so it needs to take it as parameter. 

A watched resource may use multiple Generator implementation, depending on client characteristics. Instead of adding 'if/else' in the Generator - it can be cleaner to have simpler Generate() methods. If you look a bit in the current generator code you can see what I mean:-)